ksql-engine/src/main/java/io/confluent/ksql/KsqlExecutionContext.java

```diff
@@ -88,6 +92,34 @@
    */
   PreparedStatement<?> prepare(ParsedStatement stmt);
 
+  /**
+   * Executes a query using the supplied service context.
+   */
+  TransientQueryMetadata executeQuery(
+      ServiceContext serviceContext,
+      ConfiguredStatement<Query> statement
+  );
+
+  /**
+   * Computes a plan for executing a DDL/DML statement using the engine's primary service context.
+   */
+  KsqlPlan plan(ConfiguredStatement<?> statement);
```

Comment:
I started to remove thie primary service context from this class to avoid the execute without it. I think you should remove this method in this PR.